### ISSUE TITLE
Test if environement variables *_FILE exists and is readable

### ISSUE THREAD

**Issue by camo-f** [bitnami/bitnami-docker-magento#179]
Hello,

When using environment variables mounted as files with *_FILE, there is no test to check is the given path exists and is readable.
https://github.com/bitnami/bitnami-docker-magento/blob/2.4.1-debian-10-r95/2/debian-10/rootfs/opt/bitnami/scripts/magento-env.sh#L73

It would be helpful to test and issue a warning if the file cannot be read. Main cases are :
- File mounted at wrong location
- File mounted with wrong ownership/permissions (especially when running the container as non-root)

It is rather easy to test a file in bash, with `-e` for existence, `-f` to exclude directories, and `-r` for readability.
That would end up with something like this, though I'm not sure how to manage logs.

```
for env_var in "${magento_env_vars[@]}"; do
    file_env_var="${env_var}_FILE"
    if [[ -e "${file_env_var}" ]] && [[ -f "${file_env_var}" ]] && [[ -r "${file_env_var}" ]]; then
        if [[ -n "${!file_env_var:-}" ]]; then
            export "${env_var}=$(< "${!file_env_var}")"
            unset "${file_env_var}"
        fi
    else
        echo "Warning: ${file_env_var} is not readable"
    fi
done
```

What do you think ?

Thanks

**Comment by Mauraza** [comment]
Hi @camo-f,

with this line `if [[ -n "${!file_env_var:-}" ]]; then` we check if the file exists and this script will be executed in the Dockerfile 
https://github.com/bitnami/bitnami-docker-magento/blob/909a3ced89e0b7bcc048d42f9ec6e1fc2a2b2b26/2/debian-10/Dockerfile#L29
And in this case, the user is `root`
https://github.com/bitnami/bitnami-docker-magento/blob/909a3ced89e0b7bcc048d42f9ec6e1fc2a2b2b26/2/debian-10/Dockerfile#L51

But in any case,  do you have a scenario, or has anything happened to you because of that?

**Comment by camo-f** [comment]
Hi @Mauraza,

Thank you for your clarification. Maybe I extrapolated a bit too much, so here is the scenario where I got an issue :
I ran the image as non-root (`user: 1001` in my `docker-compose.yml`) and referenced a mounted file owned by `root` in an environment variable.
So, the final value was empty, because the file wasn't readable for user `1001`.

Of course, it is my mistake to mount a file with wrong ownership. `root` is the default one, that's why I didn't get it right straightaway.

My only intent here is the improvement of the user experience, by issuing a warning when a file is not readable, whatever the reason is (missing file or wrong permission).

**Comment by Mauraza** [comment]
Hi @camo-f, 

could you tell me how you ran an image as non-root? 
If you create a Dockerfile similar to this :arrow_down: to build a container and use this new container into your `docker-compose.yaml` you should not have this problem
```
FROM docker.io/bitnami/magento:2-debian-10
USER 1001
```

A good way to detect these errors is to run `exec` using `-u root`, if we suppose the name of the container is magento:
```
docker exec -ti -u root magento bash
```
in this way, you can access into the container as a **root**, and you can check the permission of any directory.

Anyway, we evaluate if add the `-r`.

Thanks

**Comment by camo-f** [comment]
I used the `user` option of docker-compose. Then I mounted a Docker Swarm secret for `MAGENTO_DATABASE_PASSWORD_FILE`. I guess it would work the same with a volume.

```
version: '3.8'
 
services:
  magento:
    image: bitnami/magento:2.4.1
    deploy:
      replicas: 1
    user: "1001"
    environment:
      - MAGENTO_DATABASE_HOST=mysql
      - MAGENTO_DATABASE_NAME=magento_db
      - MAGENTO_DATABASE_USER=magento
      - MAGENTO_DATABASE_PASSWORD_FILE=/run/secrets/mysql-password
    secrets:
      - source: mysql-password
        target: /run/secrets/mysql-password
        mode: 0400
#        uid: '1001'
#        gid: '1001'

secrets:
  mysql-password:
    external: true
    name: magento-mysql-password-v1.0
```

I get the following logs :
```
/opt/bitnami/scripts/magento-env.sh: line 75: /run/secrets/mysql-password: Permission denied
magento 16:29:06.35 
magento 16:29:06.35 Welcome to the Bitnami magento container
magento 16:29:06.35 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-magento
magento 16:29:06.35 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-magento/issues
magento 16:29:06.35 
magento 16:29:06.35 INFO  ==> ** Starting Magento setup **
magento 16:29:06.37 INFO  ==> Configuring PHP options
magento 16:29:06.38 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
magento 16:29:06.45 WARN  ==> Hostname mysql could not be resolved, this could lead to connection issues
magento 16:29:06.55 WARN  ==> Hostname elasticsearch could not be resolved, this could lead to connection issues
magento 16:29:06.55 ERROR ==> The MAGENTO_DATABASE_PASSWORD environment variable is empty or not set. Set the environment variable ALLOW_EMPTY_PASSWORD=yes to allow a blank password. This is only recommended for development environments.
```

It says `MAGENTO_DATABASE_PASSWORD` is empty because `/run/secrets/mysql-password` is owned by root with 0400 permission, so it cannot be read by user `1001`.

If I add `uid` and `gid` in `docker-compose.yml` (commented above), the file is read and the `MAGENTO_DATABASE_PASSWORD` is populated.

In short, adding the `-r` option would simply check if the file is readable by the current user, then a warning could be issued if the test fails.

Thank you

**Comment by Mauraza** [comment]
Hi @camo-f,

could you create a pull request adding the check if is readability the file? 

Thanks!!
